src/basicio.cpp

```diff
@@ -481,47 +482,47 @@ namespace Exiv2 {
         if (hFd == INVALID_HANDLE_VALUE) {
 #ifdef EXV_UNICODE_PATH
             if (p_->wpMode_ == Impl::wpUnicode) {
-                throw WError(kerCallFailed, wpath(), "MSG1", "_get_osfhandle");
+                throw Error(ErrorCode::kerCallFailed, ws2s(wpath()), "MSG1", "_get_osfhandle");
```

Comment:
In `basicio.cpp` I really dislike all the code duplication we have to throw an exception with either  `wpath()` or `path()`. I would remove the path argument from this call so that we can remove lot of code from this file.

In my opinion is not very useful to incluse the file path in the exception, since we normally use the `exiv2` command line tool just with one file (and therefore we know which file is being used) if that exception is thrown.

Reply:
I don't like this either, although I'm mostly annoyed by the need to explicitly call `ws2s(foo)`. That should be possible to remove, but I didn't have the time to code something up yet.